### DISCUSSION TITLE
plugin/config control over the base navigation menu

### DISCUSSION THREAD

**Discussion by tlodge** [vendurehq/vendure#2189] [Q&A] [answered]
I have been starting to think about how I would like a seller's admin screen to look when using the new vendure marketplace functionality.  I am assuming technically naive users, so would like to have control over what is displayed in the main menu on the left hand side of the menu screen (i.e. currently, catalogue, sales, customers, marketing etc).  My problem is that the visibility of the menu items on the left is controlled by roles, and so does not give me the control I desire.  

To illustrate with an example.  I would like my sellers to only be able to attach their products to a set of pre-defined facets.  In order to be able to view the facets that they can assign their products to, I must provide them with the 'ReadFacet' permission.  But this then means that the left hand menu will show the 'facets' option under the catalogue, which I don't want, as this will confuse some of my sellers.  It looks like the display of the left hand menu is hard coded into the `base-nav.component.ts` file.  Similarly I'd like the option to be able to remove the display of other menu items, but without having to disable functionality by restricting permissions.

What would be great would be either to be able to replace the left hand menu with a widget of my choice, or to have the component read a config script, so that I have finer grained control over the display of the left hand menu.  But if there are any other ways I might accomplish what I want (ouside compiling a custom admin-ui) it would be great to know!

Many thanks!

**Comment by michaelbromley** [comment, marked as answer]
Hi,

This is a good suggestion, and actually the underlying infrastructure is quite well set up to support something like this - there is a hard-coded config object that defines the menu structure, so it would be a matter of allowing you to replace/modify this default config.

**Reply by tlodge** [reply]
Thanks @michaelbromley.  At some point I'm hoping to actively contribute rather than continually ask questions about features, but thanks for your responses in the meantime.
